shm/Game.cpp

```diff
@@ -7,16 +7,16 @@ constexpr size_t distancePerDay{2};
 
 Game::Game(size_t money, size_t timeLimit, size_t finalGoal)
     : money_(money), timeLimit_(timeLimit), finalGoal_(finalGoal) {
-    auto timePtr = std::make_shared<Time>();
-    time_ = timePtr.get();
+    time_ = std::make_shared<Time>();
 
-    map_ = new Map(timePtr);
+    map_ = std::make_unique<Map>(time_);
 
-    auto shipPtr = std::make_shared<Ship>(100, 100, 10, "Best Tratwa Ever", 42, timePtr);
+    auto shipPtr = std::make_shared<Ship>(100, 100, 10, "tratwa", 42, time_);
```

Comment:
Now I think we can use name `ship` instead of `shipPtr` to make it more consistent with the rest of our code here.